### PR TITLE
fix: add validation to customerIo for undefined properties

### PR DESCRIPTION
**Pull request by ItsSudip**
## What are the changes introduced in this PR?

Fixes an uncontrolled `TypeError` in the Customer.io router transformer that crashes anonymous SCREEN events when `message.properties` is `undefined`.

The crash occurs at [`src/v0/destinations/customerio/util.js:288`](https://github.com/rudderlabs/rudder-transformer/blob/develop/src/v0/destinations/customerio/util.js#L288), where `message.properties.name` is dereferenced without optional chaining inside the anonymous-event branch of `defaultResponseBuilder`. The equivalent expression in `transform.js:102` already uses `message.properties?.name` — this duplicate in `util.js` was missing the guard.

**Fix**:
```diff
- trimmedEvName = `Viewed ${truncate(message.event || message.properties.name, 86)} Screen`;
+ trimmedEvName = `Viewed ${truncate(message.event || message.properties?.name || '', 86)} Screen`;
```

The trailing `|| ''` is required because `truncate-utf8-bytes` itself throws `Input must be string` when both `message.event` and `message.properties?.name` are absent — without the fallback, optional chaining alone would just swap one Bugsnag alert for another.

**Trigger conditions** (all four must hold):
1. `messageType === 'screen'`
2. No `userId` / `email` (anonymous-event branch)
3. `message.event` is falsy
4. `message.properties` is `undefined`

**Test coverage**: Added a regression unit test in `util.test.js`. Verified that reverting the source fix makes the new test fail with the exact production error, and re-applying the fix makes it pass.

## What is the related Linear task?

Resolves [INT-6351](https://linear.app/rudderstack/issue/INT-6351)

## Please explain the objectives of your changes below

Eliminate a production Bugsnag alert (`alert group I3YVCG2ZRHG6K`, observed on v1.132.0 against Customer.io PROD) caused by malformed anonymous SCREEN events reaching the router transformer.

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

No behavioral change for well-formed events. For the previously-crashing edge case (anonymous SCREEN with both `event` and `properties` missing), the transformer now produces a payload with `name: "Viewed  Screen"` instead of throwing — matching the spirit of the analogous guard in `transform.js:102`.

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/customerio/util.test.js [hunk 1]
```diff
@@ -347,3 +347,32 @@ describe('Unit test cases for customerio defaultResponseBuilder with userId cont
     });
   });
 });
+
+describe('Unit test cases for customerio defaultResponseBuilder anonymous SCREEN edge cases', () => {
+  it('should not crash when properties is undefined for anonymous screen event', () => {
+    const expectedOutput = {
+      rawPayload: {
+        data: undefined,
+        name: 'Viewed  Screen',
```
Comment (achettyiitr) [inline src/v0/destinations/customerio/util.test.js:356]: Why is there an extra space here? Can't we just do `Viewed Screen`, even [here](http://github.com/rudderlabs/rudder-transformer/blob/develop/src/v0/destinations/customerio/transform.js#L100-L103) as well.

File: src/v0/destinations/customerio/util.js [hunk 2]
```diff
@@ -285,7 +285,7 @@ const defaultResponseBuilder = (message, evName, userId, evType, destination, me
     // CustomerIO supports 100byte of event name for anonymous users
     if (messageType === EventType.SCREEN) {
       // 100 - len(`Viewed  Screen`) = 86
-      trimmedEvName = `Viewed ${truncate(message.event || message.properties.name, 86)} Screen`;
```
Comment (achettyiitr) [inline src/v0/destinations/customerio/util.js:288]: Also, Can we have a utility for this, so that it is not segmented in 2 places.

File: src/v0/destinations/customerio/util.test.js (hunk 1 above)
Comment (ItsSudip) [reply src/v0/destinations/customerio/util.test.js:356]: In the previous section we are not adding [Here](https://github.com/rudderlabs/rudder-transformer/blob/faf5623912bd3e016442911062586bcdcf5bed85/src/v0/destinations/customerio/transform.js#L100-L102), we are not using truncate, so the event will be `Viewed undefined Screen`. But on the fix we are using truncate so added empty string in place of undefined which is resulting extra space.

Comment (achettyiitr) [reply src/v0/destinations/customerio/util.test.js:356]: Should we fix the code to send `Viewed Screen` instead of `Viewed  Screen`? Having an extra space in the middle is confusing.

File: src/v0/destinations/customerio/util.js (hunk 2 above)
Comment (ItsSudip) [reply src/v0/destinations/customerio/util.js:288]: We can't use same utility because in one place we are using truncate and in another place we are not. If we use common util method then there will be behaviour diff.

**Review by achettyiitr — APPROVED** [review]
